### ISSUE TITLE
Adapters using JSON-based config don't seem to work

### ISSUE THREAD

**Issue by davidliang2008** [amcharts/amcharts4#649]
Under JSON-based Config section on the concepts documentation, there is a section for Adapters:
https://www.amcharts.com/docs/v4/concepts/json-config/#Adapters

It doesn't seem to work, no matter if you're using the object style or the array style.

**demo:** http://jsfiddle.net/davidliang2008/aq9Laaew/288043/

For all the working CodePen demos on Adapters section, although there is a tab to show JSON-based config, it's actually using TypeScript or JavaScript **but not JSON** to config the series, adapters, etc.

![image](https://user-images.githubusercontent.com/1124420/49672371-b3118480-fa1f-11e8-8261-a33fba6ad142.png)

**Comment by martynasma** [comment]
A few things.

1) Adapters for `tooltipText` on `PieSeries` is set not on series, but rather on `slices.template`.
2) There appears to be a bug which prevents setting adapters as an array in JSON config. It's now fixed in dev, and will be released in the next version.
3) Since array-adapters were not working it does not make a lot of difference, but the `type` should be actually `key`, as per examples in docs. If you are using strong type, you might starting getting an error if you don't update your code after upgrading.

So the code should look like this:

```
    series: [{
      // ...
      slices: {
        adapter: [{
          key: "tooltipText",
          callback: function(val) {
            return "This is a PieSeries: " + val
          }
        }]
      }
    }]
```

In the interim, you can use object-based setup, which will work right now:

```
    series: [{
      // ...
      slices: {
        adapter: {
          tooltipText: function(val) {
            return "This is a PieSeries: " + val
          }
        }
      }
    }]
```

P.S. not sure I understand your note about docs snippets. I checked those and all JSON tabs seem to be showing JSON setup. Except, yeah, it's not pure JSON since it has function references that are not supported in text-based JSON, but, since there is no way to serialize JS functions, we are settling for this "hybrid" syntax.

Hope this makes sense.

**Comment by martynasma** [comment]
Fixed in [4.0.8](https://github.com/amcharts/amcharts4/releases/tag/4.0.8).

## [4.0.8] - 2018-12-11

### Added
- `currentStep` added to `ColorSet`. Allows setting current position of the color iterator.
- Adapters added for `Sprite` properties: `inertiaOptions`, `hitOptions`, `hoverOptions`, `swipeOptions`, `keyboardOptions`, `cursorOptions`.

### Changed
- `nonScaling` of `AxisLabel` is no longer set to `true` by default, to improve performance.

### Fixed
- `ValueAxis` was not updating its scale if series was hidden and animated theme was not enabled.
- Zooming an `XYChart` with all values equal was broken.
- Series in an `XYChartScrollbar` sometimes was not visible initially.
- `ColumnSeries` was drawing 1px line for `null` value data items.
- `DateAxis`' `snapTooltip = true` was working incorrectly with `skipEmptyPeriods = true`.
- Logarithmic `ValueAxis` with axis breaks was resulting in stack overflows.
- JSON: Chaining multiple adapters in an array (e.g. `adapter: [{...}, {...}]`), was not working.
- JSON: Adding elements to `children` via array syntax was not working properly.
- JSON: `Container` elements can now be created via JSON config.
- Setting `dataFields` in `Export` was only affecting column names. Values for all data fields were still being included in export.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
